### PR TITLE
feat(UIKIT-4249): parent message update

### PR DESCRIPTION
**Pull request by bang9**
## External Contributions

This project is not yet set up to accept pull requests from external contributors.

If you have a pull request that you believe should be accepted, please contact
the Developer Relations team <developer-advocates@sendbird.com> with details
and we'll evaluate if we can setup a [CLA](https://en.wikipedia.org/wiki/Contributor_License_Agreement) to allow for the contribution.

## For Internal Contributors

[UIKIT-4249]

## Description Of Changes

Add a brief description of the changes that you have involved in this PR

## Types Of Changes

What types of changes does your code introduce to this project?
Put an `x` in the boxes that apply_

- [ ] Bugfix
- [x] New feature
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance (ex) Prettier)
- [ ] Build configuration
- [ ] Improvement (refactor code)
- [ ] Test

[UIKIT-4249]: https://sendbird.atlassian.net/browse/UIKIT-4249?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/uikit-chat-hooks/src/channel/useGroupChannelMessages/useGroupChannelMessagesWithCollection.ts [hunk 1]
```diff
@@ -150,6 +150,10 @@ export const useGroupChannelMessagesWithCollection: UseGroupChannelMessages = (s
             }
           }
         }
+
+        if (_.source === MessageEventSource.EVENT_MESSAGE_UPDATED) {
```
Comment (chrisallo) [inline packages/uikit-chat-hooks/src/channel/useGroupChannelMessages/useGroupChannelMessagesWithCollection.ts:154]: 이거 언더바 말고 context 이름으로 쓰는게 어떨까요? 나중에 모르는 사람이 보면 찾아봐야하지 않을까 싶어서 명확하게 해두는게 어떨까 싶습니다~

File: packages/uikit-react-native/src/components/GroupChannelMessageRenderer/GroupChannelMessageParentMessage.tsx [hunk 2]
```diff
@@ -27,11 +29,22 @@ type Props = {
...
 
-  // FIXME: apply theme later
+  const [parentMessage, setParentMessage] = useState(() => message);
+  const type = getMessageType(parentMessage);
+
+  useEffect(() => {
+    return groupChannelPubSub.subscribe(({ type, data }) => {
+      if (type === 'MESSAGES_UPDATED') {
+        const updatedParent = data.messages.find((it) => it.messageId === parentMessage.messageId);
+        if (updatedParent?.isFileMessage() || updatedParent?.isUserMessage()) setParentMessage(updatedParent);
```
Comment (chrisallo) [inline packages/uikit-react-native/src/components/GroupChannelMessageRenderer/GroupChannelMessageParentMessage.tsx:43]: 여기서 message type을 체크하는 이유가 있을까요?

File: packages/uikit-chat-hooks/src/channel/useGroupChannelMessages/useGroupChannelMessagesWithCollection.ts (hunk 1 above)
Comment (bang9) [reply packages/uikit-chat-hooks/src/channel/useGroupChannelMessages/useGroupChannelMessagesWithCollection.ts:154]: 넵 좋습니다

File: packages/uikit-react-native/src/components/GroupChannelMessageRenderer/GroupChannelMessageParentMessage.tsx (hunk 2 above)
Comment (bang9) [reply packages/uikit-react-native/src/components/GroupChannelMessageRenderer/GroupChannelMessageParentMessage.tsx:43]: 아 Base 메세지로 넘어와서 타입 체크때문에 해놨는데, type predicates 로 변경해보겠습니다!

**Pull request merged**